lib/src/rules/avoid_js_rounded_ints.dart

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+
+const _desc = r'Avoid JavaScript rounded ints.';
+
+const _details = r'''
+
+**AVOID** integers that will be rounded.
+
+When a program is compiled to JavaScript `int` and `double` become JavaScript
+Numbers. Too large integers (`value < Number.MIN_SAFE_INTEGER` or
+`value > Number.MAX_SAFE_INTEGER`) may be rounded to the closest Number value.
```

Comment:
Is it possible to put actual values in the message? It would make the message more concrete.
If you could say something like the following, that would make the situation clearer:

1000000000000000001 cannot be represented exactly as a JavaScript Number, so 1000000000000000000 will be used instead.

Reply:
Note that the nearest representable number is `value.toDouble()`, but this might be a value that cannot be represented as an `int` (when 0x7ff....fff is rounded up to 0x800....000), but you could print it with
`new BigInt.from(value.toDouble()).toString()`

Reply:
>Is it possible to put actual values in the message?

It's not something that linter currently does. It would need some changes if we want to do that. 
@pq @bwilkerson what do you think about customizing messages depending on the code linted ?

Reply:
Actually, I think it's possible to do that today, it's just that nobody has done it so there's no pattern to follow. I'd be fine with merging this change and then refining it, or you can make this change before merging if you'd like.

The trick is that you are not restricted to using `reportLint` to report a lint; I believe you have access to the error reporter that `reportLint` uses so you can create your own LintCode with a message that can be customized by using the substitution mechanism supported by errors. For example, the message might look something like `"The value '{0}' will be rounded to {1} when compiled to JavaScript."`.